scripts/install-nix-from-closure.sh

```diff
@@ -1,149 +1,635 @@
+#    "
+#
+###############################
+#
+# This is the best balance of code simplicity and code readability found so far.
+# This produces output in color:
+#
+###############################
+#
+#./install: Notice:
+#
+#    Install executed for ROOT.
+#
+#    Doing classic Linux package manager mode.
+#    In Nix this mode is called: single-user mode for root.
+#
+#    Nix can do multi-user mode, and manage package trees for users independently.
+#
+###############################
+#
+true # so {...} is not empty, shell will not permit otherwise
+}
+###############################
+###  Setup environment
+###############################
+{
+# Set the character collating sequence to be numeric ASCII/C standard.
+# Set the character set to be standard one-byte ASCII.
+# Set default umask; to be non-restrictive and friendly to others.
+#
+LC_COLLATE=C
+LANG=C
+umask 022
+}
+###############################
+###  Main constants
+###############################
+{
+readonly dest="/nix"
+readonly self="$(dirname "$(realpath "$0")")"
+readonly nix="@nix@"
+readonly cacert="@cacert@"
+readonly appname="$0"
+}
+###############################
+###  CLI control constants
+###############################
+{
+if test -t ; then    # File descriptor is associated with a terminal - output colors
+    if tput colors > /dev/null 2>&1 ; then
+        # use tput and terminfo DB
+        readonly red=$(tput setaf 1)
+        readonly green=$(tput setaf 2)
+        readonly yellow=$(tput setaf 3)
+        readonly blue=$(tput setaf 4)
+        readonly bold=$(tput smso)
+        readonly reset=$(tput sgr0) # Reset to default output
+    else
+        # tput is not present on some systems (Alpine Linux), this trick with printf allows to store, not 'codes' - literal symbols
+        readonly red=$(printf '\033[1;31m')
+        readonly green=$(printf '\033[1;32m')
+        readonly yellow=$(printf '\033[1;33m')
+        readonly blue=$(printf '\033[1;34m')
+        readonly bold=$(printf '\033[1m')
+        readonly reset=$(printf '\033[0;m') # Reset to default output
+    fi
+fi
+}
+###############################
+###  CLI output functions
+###############################
+{
+print() {
+    # Unified output function
+    # Using printf, because it is more portable across all platforms than echo, and it is important for Nix install
+    #
+    if [ -z "$message" ]; then
+        message="$1"
+    fi
+    if [ -z "$color" ]; then
+        color="$reset"
+    fi
+    if [ -z "$prefix" ]; then
+        prefix='Info'
+    fi
+    # This line makes all prints in script
+    # Form:
+    ### Application: Prefix: Body of message
+    printf '%s%s: %s: %s%s\n' "$color" "$appname" "$prefix" "$message" "$reset"
+
+    # Output is done, unset variables that could've been set by parent functions
+    unset message
+    unset color
+    unset prefix
+}
+
+notice() {
+    message="$1"
+    color="$green"
+    prefix='Notice'
+    print
```

Comment:
It is done to easily change/extend any print functions.
Otherwise, if I would decide to add new feature to them, change their form - I would need to go redo all the code lines where they are used.